src/jvmTest/java/it/krzeminski/snakeyaml/engine/kmp/comments/EmitterWithCommentEnabledTest.kt

```diff
@@ -0,0 +1,639 @@
+/*
+ * Copyright (c) 2018, SnakeYAML
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
```

Comment:
question: since this is no longer the original code do we need to keep the original copyright?

Reply:
I was wondering the same, but since we have this preamble even in the production (non-test) code, I decided to follow this pattern. We can discuss it and remove it in the whole repo as a follow-up. @aSemy thoughts?

Reply:
Sorry, I don't know :)

Reply:
Looks like you should keep it... Quote from Apache license:

> You must retain, in the Source form of any Derivative Works that You distribute, all copyright, patent, trademark, and attribution notices from the Source form of the Work, excluding those notices that do not pertain to any part of the Derivative Works;

And probably this file can be described as Derivative Work

Reply:
Right, but we may want to add us there as well. I created https://github.com/krzema12/snakeyaml-engine-kmp/issues/271 to follow up on it.